File: tests/fixtures/go_repo/internal/types/item_dto.go
// Package types — request/response shapes for /items endpoints.
package types

import "time"

// CreateItemRequest is the JSON body for POST /items.
type CreateItemRequest struct {
	Title       string `json:"title" validate:"required,min=1,max=200"`
	Description string `json:"description" validate:"max=2000"`
}

// UpdateItemRequest is the JSON body for PATCH /items/{id}. Optional fields.
type UpdateItemRequest struct {
	Title       *string `json:"title,omitempty" validate:"omitempty,min=1,max=200"`
	Description *string `json:"description,omitempty" validate:"omitempty,max=2000"`
}

// ItemResponse is the public, serialised shape of an Item.
type ItemResponse struct {
	ID          string    `json:"id"`
	OwnerID     string    `json:"owner_id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
}

// ItemListResponse wraps a page of items with pagination metadata.
type ItemListResponse struct {
	Items      []ItemResponse `json:"items"`
	TotalCount int            `json:"total_count"`
	Page       int            `json:"page"`
	PageSize   int            `json:"page_size"`
}

File: tests/fixtures/rust_repo/src/dto/item_dto.rs
//! Request / response types for `/items` endpoints.

use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;
use validator::Validate;

/// JSON body for `POST /items`. `description` is optional and defaults to
/// the empty string when omitted.
#[derive(Debug, Clone, Deserialize, Validate)]
pub struct CreateItemRequest {
    #[validate(length(min = 1, max = 200))]
    pub title: String,
    #[serde(default)]
    pub description: String,
}

/// JSON body for `PATCH /items/{id}` — both fields optional.
#[derive(Debug, Clone, Deserialize, Validate)]
pub struct UpdateItemRequest {
    #[validate(length(min = 1, max = 200))]
    pub title: Option<String>,
    pub description: Option<String>,
}

/// Public-facing item shape returned by every read endpoint.
#[derive(Debug, Clone, Serialize)]
pub struct ItemResponse {
    pub id: Uuid,
    pub owner_id: Uuid,
    pub title: String,
    pub description: String,
    pub created_at: DateTime<Utc>,
}

/// Paginated wrapper used by `GET /items`.
#[derive(Debug, Clone, Serialize)]
pub struct ItemListResponse {
    pub items: Vec<ItemResponse>,
    pub total_count: i64,
    pub page: u32,
    pub page_size: u32,
}

impl From<crate::models::Item> for ItemResponse {
    fn from(i: crate::models::Item) -> Self {
        ItemResponse {
            id: i.id,
            owner_id: i.owner_id,
            title: i.title,
            description: i.description,
            created_at: i.created_at,
        }
    }
}
